includes/class-events-store-cron-filters.php

```diff
@@ -0,0 +1,264 @@
+<?php
+/**
+ * Event storage for WP 5.1 and later.
+ *
+ * @package a8c_Cron_Control
+ */
+
+namespace Automattic\WP\Cron_Control;
+
+/**
+ * Class Option_Intercept
+ */
+class Events_Store_Cron_Filters extends Singleton {
+	/**
+	 * Register hooks to intercept events before storage.
+	 */
+	protected function class_init() {
+		add_filter( 'pre_schedule_event', [ $this, 'filter_event_scheduling' ], 10, 2 );
+		add_filter( 'pre_reschedule_event', [ $this, 'filter_event_rescheduling' ], 10, 2 );
+		add_filter( 'pre_unschedule_event', [ $this, 'filter_event_unscheduling' ], 10, 4 );
+		add_filter( 'pre_clear_scheduled_hook', [ $this, 'filter_clear_scheduled_hook' ], 10, 3 );
+		add_filter( 'pre_unschedule_hook', [ $this, 'filter_unchedule_hook' ], 10, 2 );
+		add_filter( 'pre_get_scheduled_event', [ $this, 'filter_event_retrieval' ], 10, 4 );
+		add_filter( 'pre_next_scheduled', [ $this, 'filter_next_scheduled' ], 10, 4 );
```

Comment:
This filter was removed in WordPress/wordpress-develop@554ee5b53c2c16ec4dc7255b5839205b8236dbf1 and the logic for finding the next scheduled event moved to `wp_get_scheduled_event()` which can be hijacked using the `pre_get_scheduled_event` filter.